### ISSUE TITLE
Got 9.0.0 throws when using unescaped characters

### ISSUE THREAD

**Issue by lostfictions** [sindresorhus/got#563]
With Got 9.0.0,
```
require("got")("http://google.com/search", { query: { q: "friend’s house" } })
   .then(() => console.log("ok"))
```
rejects with:
```
(node:27956) UnhandledPromiseRejectionWarning: RequestError: Request path contains unescaped characters
    at EventEmitter.cacheReq.on.error (node_modules/got/source/request-as-event-emitter.js:105:27)
    at EventEmitter.emit (events.js:182:13)
    at makeRequest (node_modules/cacheable-request/src/index.js:125:9)
    at get (node_modules/cacheable-request/src/index.js:135:14)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

With `got@8.3.2` this snippet works just fine.

**Comment by jstewmon** [comment]
@lostfictions I'm not able to reproduce your issue (see #564). What version of node are you using?

**Comment by lostfictions** [comment]
That's odd -- I was also able to repro it on another system. I notice Got seems to not be using a lockfile or shrinkwrap and specifies version ranges instead of exact versions to boot -- maybe some dependency or transitive dependency broke something? Here's a transcription of a full session where I install from scratch and run the line of code above -- it repros immediately for me. Node version is 10.6.0, as verified below. I've also included the output of `npm ls` in case that's helpful in debugging.

```
~/Code$ mkdir temp && cd temp

~/Code/test$ npm i got
npm WARN saveError ENOENT: no such file or directory, open 'test/package.json'
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN enoent ENOENT: no such file or directory, open 'test/package.json'
npm WARN testt No description
npm WARN testt No repository field.
npm WARN testt No README data
npm WARN testt No license field.

+ got@9.0.0
added 19 packages from 7 contributors and audited 23 packages in 1.832s
found 0 vulnerabilities

~/Code/test$ node -e 'require("got")("http://google.com/search", { query: { q: "friend’s house" } }).then(() => console.log("ok"))'
(node:16167) UnhandledPromiseRejectionWarning: RequestError: Request path contains unescaped characters
    at EventEmitter.cacheReq.on.error (node_modules/got/source/request-as-event-emitter.js:105:27)
    at EventEmitter.emit (events.js:182:13)
    at makeRequest (node_modules/cacheable-request/src/index.js:125:9)
    at get (node_modules/cacheable-request/src/index.js:135:14)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:16167) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:16167) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

~/Code/test$ node --version
v10.6.0

~/Code/test$ npm ls
└─┬ got@9.0.0
  ├─┬ @sindresorhus/is@0.11.0
  │ └── symbol-observable@1.2.0
  ├─┬ cacheable-request@4.0.1
  │ ├─┬ clone-response@1.0.2
  │ │ └── mimic-response@1.0.1 deduped
  │ ├── get-stream@3.0.0 deduped
  │ ├── http-cache-semantics@4.0.0
  │ ├─┬ keyv@3.0.0
  │ │ └── json-buffer@3.0.0
  │ ├── lowercase-keys@1.0.1
  │ ├── normalize-url@3.2.0
  │ └─┬ responselike@1.0.2
  │   └── lowercase-keys@1.0.1 deduped
  ├─┬ decompress-response@3.3.0
  │ └── mimic-response@1.0.1 deduped
  ├── duplexer3@0.1.4
  ├── get-stream@3.0.0
  ├── mimic-response@1.0.1
  ├── p-cancelable@0.5.0
  ├── to-readable-stream@1.0.0
  └─┬ url-parse-lax@3.0.0
    └── prepend-http@2.0.0

~/Code/test$ uname -rv
4.15.0-30-generic #32~16.04.1-Ubuntu SMP Thu Jul 26 20:25:39 UTC 2018
```

**Comment by szmarczak** [comment]
It's natural `http` behavior:

```
> require('http').request({path:'friend’s house'})
TypeError [ERR_UNESCAPED_CHARACTERS]: Request path contains unescaped characters
    at new ClientRequest (_http_client.js:105:13)
    at Object.request (http.js:41:10)
```

but IMO it'd be very nice to support it.

**Comment by lostfictions** [comment]
@szmarczak As I noted, it's a regression from previous behaviour. I unfortunately don't have the capacity to deal with regressions in basic behaviour like this so I've already migrated from Got to a better-tested/more popular library and am unsubscribing from this issue, but I hope you fix it!

**Comment by szmarczak** [comment]
> I unfortunately don't have the capacity to deal with regressions in basic behaviour like this so I've already migrated from Got to a better-tested/more popular library and am unsubscribing from this issue

I'm very sorry to hear that 😢 @jstewmon has fixed that in #564 which I believe will be merged ASAP.
